migrations/2023-07-05-000058_person-admin/up.sql

```diff
@@ -0,0 +1 @@
+create index idx_person_admin on person(admin) where admin; -- allow quickly finding all admins (PersonView::admins)
```

Comment:
Please remove the migration as its already in https://github.com/LemmyNet/lemmy/pull/3479

Reply:
this one is better than the other one because it only includes the admin part. the other one means pg has to store a copy of the key (8byte) + the admin bool (maybe another byte or 8) for **all** persons. my index just requires it to store them for admin_count persons. this index also does not have to be updated if persons change that are not admins which makes it faster.

at the level of fruits hanging off the lemmy tree, that's probably not a too relevant distinction though.

Reply:
Okay then you can change it in the existing migration and remove this one.

Reply:
the filter restricts which rows are actually contained of the index, called a partial index. it only works for boolean filters (= and !=), not ordered queries (<= or >=). filtered indexes are very useful to keep index size small when you're only looking at a small subset of data and at the same time reduce updating overhead.

the disadvantage is that it is also only usable for positive queries i.e. this index cannot be used to find users that are **not** an admin. But since 99% of users are not admins, that's a useless index anyways.

docs: https://www.postgresql.org/docs/current/indexes-partial.html

Reply:
Cool, thanks. There might be some other boolean columns that could be optimized, or even the hot rank ones.